js/event.js

```diff
@@ -1,29 +1,119 @@
+
+const collectFormData = () => {
+
+    const formData = new FormData()
+
+    const inputData = {}
+    const inputs = document.querySelectorAll('[data-key]')
 
-const eventButtons = document.querySelectorAll('#eventModal button')
+    inputs.forEach(input => {
+        if(input.required && !input.value){
+            throw(`Required field (${input.dataset.key}) skipped`)
+        }
+        const key = input.dataset.key
+        const value = input.value
+        inputData[key] = value ? value : null
+    })
+
+    formData.title = inputData.title
+
+    const { startTime : startTimeString, endTime : endTimeString} = inputData
+
+    if(inputData.date && startTimeString && endTimeString){
+
+        const startTime = parseTimeString(startTimeString)
+        const startDate = new Date(inputData.date)
+        startDate.setHours(startTime.hour, startTime.minutes)
+        formData.startDate = Date.parse(startDate)
+    
+        const endTime = parseTimeString(endTimeString)
+        const endDate = new Date(inputData.date)
+        endDate.setHours(endTime.hour, endTime.minutes)
+        formData.endDate = Date.parse(endDate)
+
+        formData.duration = Math.ceil( (formData.endDate - formData.startDate) / (MILISECOND_HOUR/4) )
+        formData.timeSlot = Math.floor( startTime.minutes /  TIMESLOT_LENGTH)
+        formData.cellTimestamp = startDate.setMinutes(0).valueOf()
+    }
 
-eventButtons.forEach(btn => {
-    btn.addEventListener('click', (e) => e.preventDefault())
+    formData.setId()
+    formData.createdAt = getToday().valueOf()
+    formData.description = inputData.description
+
+    localStorage.setItem(formData.id, JSON.stringify(formData))
+    storageState.setState( formData )
+    
+}
+
+eventButton_save.addEventListener('click', e => {
+    try{
+        collectFormData()
+        toggleDisplay()
+        e.preventDefault()
+    }
+    catch(e){
+        console.error(e)
+    }
 })
 
+const setFormInputValues = (date) => {
+    const time = date.toTimeString().slice(0, 5)
+    //use 'lt-LT' as locale to correctly form date as YYYY-MM-DD
+    const YMDdate = date.toLocaleDateString('lt-LT', {year: 'numeric', month: 'numeric', day: 'numeric'})
+    eventDate.value = YMDdate
+    startTime.value = time
+    endTime.value = time
+}
+
 eventButtons_create.forEach(btn => {
-    btn.addEventListener('click', () => toggleDisplay())
+    btn.addEventListener('click', (e) => {
+        displayModal(e, getToday() )
+    })
 })
 
-eventButton_cancel.addEventListener('click', () => toggleDisplay())
-eventButton_save.addEventListener('click', () => toggleDisplay())
+export const displayModal = (e, date) => {
+    toggleDisplay()
+    setFormInputValues( date )
+}
 
-const toggleDisplay = () => {
-    const eventModal = document.querySelector("#eventModal")
-    if(eventModal.classList.contains('display-none')){
-        eventModal.classList.remove('display-none')
-    }
-    else{
+eventButton_cancel.addEventListener('click', (e) => {
```

Comment:
It is a good idea to avoid code which would execute as soon as file is imported. You could call some function to get all events attached when you need it.